### PR TITLE
Lockup module v2

### PR DESCRIPTION
**Pull request by antstalepresh**

### CODE REVIEW

File: proto/osmosis/lockup/query.proto [hunk 1]
```diff
@@ -33,6 +35,8 @@ service Query {
 
 	// Returns account locked records with longer duration
 	rpc AccountLockedLongerThanDuration(AccountLockedLongerDurationRequest) returns (AccountLockedLongerDurationResponse);
+	// Returns account locked records with longer duration excluding tokens started unlocking
+	rpc AccountLockedLongerDurationNotUnlockingOnly(AccountLockedLongerDurationNotUnlockingOnlyRequest) returns (AccountLockedLongerDurationNotUnlockingOnlyResponse);
```
Comment (sunnya97) [inline proto/osmosis/lockup/query.proto:39]: Let's either add in word `Than` to this function name, or remove it from the above one

File: proto/osmosis/lockup/tx.proto [hunk 2]
```diff
@@ -12,8 +12,12 @@ option go_package = "github.com/c-osmosis/osmosis/x/lockup/types";
 service Msg {
     // LockTokens lock tokens
     rpc LockTokens(MsgLockTokens) returns (MsgLockTokensResponse);
+	// BeginUnlockTokens begin unlocking all tokens
+	rpc BeginUnlockTokens(MsgBeginUnlockTokens) returns (MsgBeginUnlockTokensResponse);
```
Comment (sunnya97) [inline proto/osmosis/lockup/tx.proto:16]: Let's call this `BeginUnlocking`

File: x/lockup/keeper/iterator.go [hunk 3]
```diff
@@ -49,71 +49,76 @@ func (k Keeper) iterator(ctx sdk.Context, prefix []byte) sdk.Iterator {
 }
 
 // LockIteratorAfterTime returns the iterator to get locked coins
-func (k Keeper) LockIteratorAfterTime(ctx sdk.Context, time time.Time) sdk.Iterator {
-	return k.iteratorAfterTime(ctx, types.KeyPrefixLockTimestamp, time)
+func (k Keeper) LockIteratorAfterTime(ctx sdk.Context, unlockingPrefix []byte, time time.Time) sdk.Iterator {
```
Comment (sunnya97) [inline x/lockup/keeper/iterator.go:52]: Shouldn't these be `bool`s instead?  Callers of these functions shouldn't have to know what Prefixes and stuff it uses.

File: x/lockup/keeper/lock.go [hunk 4]
```diff
@@ -65,42 +79,89 @@ func (k Keeper) GetModuleBalance(ctx sdk.Context) sdk.Coins {
...
+	notUnlockingCoins := k.getCoinsFromIterator(ctx, k.LockIterator(ctx, types.KeyPrefixNotUnlocking))
+	unlockingCoins := k.getCoinsFromIterator(ctx, k.LockIteratorAfterTime(ctx, types.KeyPrefixUnlocking, ctx.BlockTime()))
+	return notUnlockingCoins.Add(unlockingCoins...)
 }
 
 // GetAccountUnlockableCoins Returns whole unlockable coins which are not withdrawn yet
 func (k Keeper) GetAccountUnlockableCoins(ctx sdk.Context, addr sdk.AccAddress) sdk.Coins {
-	return k.getCoinsFromIterator(ctx, k.AccountLockIteratorBeforeTime(ctx, addr, ctx.BlockTime()))
+	return k.getCoinsFromIterator(ctx, k.AccountLockIteratorBeforeTime(ctx, types.KeyPrefixUnlocking, addr, ctx.BlockTime()))
 }
```
Comment (sunnya97) [inline x/lockup/keeper/lock.go:92]: Let's have a `GetAccountUnlockingCoins`

**Comment by sunnya97** [comment]
Shouldn't we add a `IsUnlocking` bool to the PeriodLock type?  That way when you query a list of PeriodLocks you can differentiate between the two types?

File: proto/osmosis/lockup/query.proto (hunk 1 above)
Comment (antstalepresh) [reply proto/osmosis/lockup/query.proto:39]: I would like to keep names shorter, removed Than from above function

File: proto/osmosis/lockup/tx.proto (hunk 2 above)
Comment (antstalepresh) [reply proto/osmosis/lockup/tx.proto:16]: Ok

File: x/lockup/keeper/iterator.go (hunk 3 above)
Comment (antstalepresh) [reply x/lockup/keeper/iterator.go:52]: Ok, bool makes more sense.

File: x/lockup/keeper/lock.go (hunk 4 above)
Comment (antstalepresh) [reply x/lockup/keeper/lock.go:92]: UnlockingCoins should include unlockable coins as well or only unlocking started but not unlockable coins?

**Comment by antstalepresh** [comment]
> Shouldn't we add a `IsUnlocking` bool to the PeriodLock type? That way when you query a list of PeriodLocks you can differentiate between the two types?

For not unlocking coins, EndTime is unset and it could be used as flag I think.

File: x/lockup/keeper/lock.go (hunk 4 above)
Comment (sunnya97) [reply x/lockup/keeper/lock.go:97]: The second one:  `only unlocking started but not unlockable coins`

File: proto/osmosis/lockup/query.proto (hunk 1 above)
Comment (sunnya97) [reply proto/osmosis/lockup/query.proto:41]: Now the keeper functions have the word "Than" it them (i.e. https://github.com/osmosis-labs/osmosis/pull/41/files#diff-eaa0d5418b72ff86c277ac9083dc98ac3cd54dcea1aaca7a51a134dd4133da21R131), but the rpc functions do not.  Should we unify these?

Comment (sunnya97) [reply proto/osmosis/lockup/query.proto:41]: Fixed it

**Pull request merged**